frontend/src/common/utils/api/handler.ts

```diff
@@ -1,20 +1,34 @@
 import type { NextApiRequest, NextApiResponse } from "next";
-import readJson from "./readJson";
+import HTTP_STATUS_CODE from "src/common/constants/HTTP_STATUS_CODE";
+import { readJson } from "src/common/utils/api/readJson";
 
-const getHandler = (path: string) => {
-  const data = readJson(path);
+const getHandler = (filePath: string) => {
+  const data = readJson(filePath);
 
-  async function handler(req: NextApiRequest, res: NextApiResponse) {
+  function handler(req: NextApiRequest, res: NextApiResponse) {
     try {
       const entityId = req.query.entityId;
-      if (data[entityId as keyof typeof data])
-        res.status(200).json(data[entityId as keyof typeof data]);
-      else res.status(204);
+      const response = data[entityId as keyof typeof data];
+
+      if (response) {
+        res.status(HTTP_STATUS_CODE.OK).json(response);
+      } else {
+        /**
+         * (thuang): Setting .status() is not enough. We still need
+         * to send the response with .send() or .json().
+         * Otherwise, the response will hang and the client will
+         * wait forever with a pending request.
+         */
+        res.status(HTTP_STATUS_CODE.NO_CONTENT).send(null);
```

Comment:
this needs to end with `send()` or `json()` in order for Next.js to resolve the request